bot/cogs/core/paginator.py

```diff
@@ -11,13 +11,19 @@
 
 
 class PaginatorView(discord.ui.View):
-    def __init__(self, get_page, page_count, timeout):
+    def __init__(self, author_id, get_page, page_count, timeout):
         super().__init__(timeout=timeout)
 
+        self._author_id = author_id
         self._page = 0
         self._page_count = page_count
         self._get_page = get_page
 
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        if interaction.user.id != self._author_id:
+            return False
+        return True
```

Comment:
this can be simplified to :)
```suggestion
        return interaction.user.id == self._author_id
```